scripts/PyChop/ISISDisk.py

```diff
@@ -0,0 +1,384 @@
+            self.nslot = [6, 1, 2, 6, 2]                       # number of slots in each chopper. Assumed equally spaced
+            self.guide_width = [40, 40, 40, 40, 20]            # width of the guide in mm
+            self.radius = [290, 545, 290, 290, 290]            # radius in mm of each disk at centre of window
+            self.numDisk = [2, 1, 1, 1, 2]                     # whether double or single disks
+            # possible instname: ['LET', 'LETHIFLUX', 'LETINTERMED', 'LETHIRES'] - corresponds to different configurations
+            if 'FLUX' in instname or (variant is not None and 'FLUX' in variant.upper()):
+                self.slot_width = [40, 890, 56, 52, 31]        # width of chopper slots in mm
+                self.variant = 'High flux'
+            elif 'RES' in instname or (variant is not None and 'RES' in variant.upper()):
+                self.slot_width = [40, 890, 56, 52, 20]        # width of chopper slots in mm
+                self.variant = 'High resolution'
+            else:
+                self.slot_width = [40, 890, 56, 52, 20]        # width of chopper slots in mm
+                self.variant = 'Intermediate'
+            self.ph_ind = 1        # index of chopper with user-determined phase
+            self.samp_det = 3.5    # sample to detector distance in m
+            self.chop_samp = 1.5   # final chopper to sample distance
+            self.source_rep = 10   # rep rate of source
+            self.tmod = 3500       # maximimum emmision window from moderator in us
+            self.frac_ei = 0.90    # fraction of Ei to plot energy loss lines
+            self.Chop2Phase = 5    # Phase delay time in usec for chopper 2 (T0/frame overlap chopper)
+        elif 'MERLIN' in instname:
+            self.dist = [9.3, 10.1]
+            self.nslot = [1, 2]
+            self.slot_width = [950, 10]
+            self.guide_width = [64, 10]
+            self.radius = [250, 290]
+            self.numDisk = [1, 1]
+            self.samp_det = 2.5
+            self.chop_samp = 1.82
+            self.source_rep = 50
+            self.tmod = 50
+            self.frac_ei = 0.90
+            self.instname = 'MERLIN'
+            self.variant = 'G'
+            self.ph_ind = 0
+            self.Chop2Phase = 1500 # Phased to not let neutrons with Ei>200meV through
+        else:
+            raise ValueError('Instrument %s not recognised.' % (instname))
+
+    def setChopper(self, variant):
+        """
+        Sets instrument variant from name
+        """
+        self.setInstrument(self.instname, variant)
+
+    def getChopper(self):
+        """
+        Returns the instrument variant name
+        """
+        return self.variant
+
+    def setFrequency(self, frequency, **kwargs):
+        """
+        Sets the chopper frequencies, in Hz.
+        If scalar, sets the resolution chopper freq to this and the pulse remover to freq/2
+        """
+        if 'LET' in self.instname:
```

Comment:
You're right that it should be reorganised. I don't really like it that at the moment there are almost entirely separate branches for LET, and for MERLIN/MAPS/MARI. But, LET was enough different that it would require rewritting a lot of the legacy code for the other instruments to accomodate it (the code for the other instruments is essentially a port of a FORTRAN program - so should probably also be rewritten). 

We have future plans to discuss with Doug and other SNS instrument scientists about a unified framework, which can also handle CNCS/ARCS/SEQUOIA (probably not HYSPEC, but possibly also including it). 

Also, the code for MERLIN/MAPS/MARI can't calculate multiple-Ei, which will be an option on all three instruments in future.

So, I'll like change this tool in the next release, but in the mean time I would like to push something which is underneath the same as the tools currently used, but with a better UI.